### DISCUSSION TITLE
Smart filters examples

### DISCUSSION THREAD

**Discussion by lumib95** [provectus/kafka-ui#1930] [Q&A] [answered]
Hi everyone, 

I have been searching in google and in the issues section but with no luck.
We do not use a schema registry and our data messages are in json format.

i tried this https://github.com/provectus/kafka-ui/pull/1868 but with no luck.

**Comment by Haarolean** [comment, marked as answer]
Hey, thanks for reaching out.

We still haven't done any documentation on smart filters but we're planning to (#1865).

Right now, you can refer to this:

variables bind to groovy context: partition, timestampMs, keyAsText, valueAsText, header, key (json if possible), value (json if possible)

json parsing logic:
for key and value if they can be parsed to json they bound as json objects, if not - bound as null

sample scripts:

1. ` keyAsText != null && keyAsText ~"([Gg])roovy" ` - regex on key as a string
2. ` value.name == 'iliax' && value.age > 30  `  - when value is json 
3.  `value == null && valueAsText != null` - search for values that are not null and are not json
4. `headers.sentBy == 'some system' && headers['sentAt'] == '2020-01-01' `
5. multiline filters also allowed:
```
def name = value.name  
def age = value.age
name == 'iliax' && age == 30
```

**Reply by nelisSpotOn** [reply]
Is there a way to search for an ID that is nested in an array? For example I have this:

```json
{
    "transaction_id": 1234,
    "events": [
        {
            "event_id": 2345
        },
        {
            "event_id": 3456
        }
    ]
}
```

I would like to get back any messages that have the `event_id` 3456 in their event array.

**Reply by Haarolean** [reply]
@nelisSpotOn #4255 -> https://github.com/kafbat/kafka-ui
